packages/babel-plugin-transform-vite-meta-glob/src/index.ts

```diff
@@ -52,6 +52,77 @@ export default function viteMetaGlobBabelPlugin({
 
           path.replaceWith(replacement)
         }
+      },
+      VariableDeclaration(path, state) {
+        const id = path.node.declarations[0].id
+        const call = path.node.declarations[0].init
+        const callee = t.isCallExpression(call) && t.isMemberExpression(call.callee) && call.callee
+        const identifier = t.isIdentifier(id) && t.identifier(id.name)
+
+        if (!identifier || !callee) {
+          return
+        }
+
+        const args = call.arguments
+        const sourceFile = state.file.opts.filename
+        const propertyName = t.isIdentifier(callee.property) && callee.property.name
+        const eagerOption =
+          t.isObjectExpression(args[1]) &&
+          args[1].properties.filter(
+            (p) => t.isObjectProperty(p) && t.isIdentifier(p.key) && p.key.name === 'eager'
+          )
+
+        if (!sourceFile || !propertyName || !eagerOption || eagerOption.length === 0) {
+          return
+        }
+
+        if (
+          isGlobKey(propertyName) &&
+          t.isMetaProperty(callee.object) &&
+          args.length === 2 &&
+          t.isStringLiteral(args[0]) &&
+          t.isObjectProperty(eagerOption[0]) &&
+          t.isBooleanLiteral(eagerOption[0].value)
+        ) {
+          const cwd = nodePath.dirname(sourceFile)
+          const globPaths = glob.sync(args[0].value, { cwd })
+
+          if (eagerOption[0].value.value) {
+            const identifiers = globPaths.map((_, idx) => t.identifier(`__glob__0_${idx}`))
+
+            const imports = globPaths.map((globPath, idx) => {
+              const specifier = t.importNamespaceSpecifier(identifiers[idx])
+              const modulePath = t.stringLiteral(globPath)
+              return t.importDeclaration([specifier], modulePath)
+            })
+
+            const variable = t.variableDeclaration('const', [
+              t.variableDeclarator(
+                identifier,
+                t.objectExpression(
+                  globPaths.map((globPath, idx) =>
+                    t.objectProperty(t.stringLiteral(globPath), identifiers[idx])
+                  )
+                )
+              )
+            ])
+
+            path.replaceWithMultiple([...imports, variable])
+          } else {
+            const replacement = t.variableDeclaration('const', [
+              t.variableDeclarator(
+                identifier,
+                t.objectExpression(
+                  globPaths.map((globPath) =>
+                    t.objectProperty(t.stringLiteral(globPath), asts[propertyName](globPath))
+                  )
+                )
+              )
+            ])
+
+            path.replaceWith(replacement)
+          }
+        }
```

Comment:
This can def be tidied up, but will wait until we're sure that this is producing the right/desired result before doing so